kyo-core/shared/src/test/scala/kyoTest/fibersTest.scala

```diff
@@ -150,26 +150,44 @@ class fibersTest extends KyoTest:
             IOs.toTry(Fibers.runAndBlock(Duration.Infinity)(
                 Fibers.timeout(10.millis)(Fibers.sleep(1.day).andThen(1))
             )).map {
-                case Failure(Fibers.Interrupted) => succeed
-                case v                           => fail(v.toString())
+                case Failure(_: Fibers.Interrupted) => succeed
+                case v                              => fail(v.toString())
             }
         }
 
         "block timeout" in runJVM {
             IOs.toTry(Fibers.runAndBlock(10.millis)(
                 Fibers.sleep(1.day).andThen(1)
             )).map {
-                case Failure(Fibers.Interrupted) => succeed
-                case v                           => fail(v.toString())
+                case Failure(_: Fibers.Interrupted) => succeed
+                case v                              => fail(v.toString())
             }
         }
 
         "multiple fibers timeout" in runJVM {
             IOs.toTry(Fibers.runAndBlock(10.millis)(
                 Seqs.fill(100)(Fibers.sleep(1.milli)).unit.andThen(1)
             )).map {
-                case Failure(Fibers.Interrupted) => succeed
-                case v                           => fail(v.toString())
+                case Failure(_: Fibers.Interrupted) => succeed
+                case v                              => fail(v.toString())
+            }
+        }
+    }
+
+    "timeout" - {
+
+        "success" in run {
+            for
+                result <- Fibers.timeout(100.millis)(Fibers.init(42).map(_.get))
+            yield assert(result == 42)
+        }
+
+        "failure" in run {
+            IOs.attempt(Fibers.timeout(1.millis)(Fibers.sleep(100.millis))).map {
+                case Failure(Fibers.Interrupted(trace)) =>
+                    assert(trace.snippet == "timeout(1.millis)(Fibers.sleep(100.millis))")
```

Comment:
Checking only the snippet so the test doesn't start failing in case the line number in the `Trace` changes due to the test scenario changing position in the file.